dbreader/dbreader.go

```diff
@@ -0,0 +1,85 @@
+package dbreader
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ppc64le-cloud/kube-rtas/knode"
+
+	"k8s.io/klog/v2"
+
+	// SQLite driver.
+	_ "rsc.io/sqlite"
+)
+
+type DbReader interface {
+	ReadServicelogDB()
+}
+
+// ServiceLog is a template to retrieve relevant fields from the servicelog.db.
+// This can further be extended to retrieve other fields that are available in the servicelog DB.
+type ServiceLog struct {
+	Id   int    `json:"Id"`
+	Desc string `json:"Description"`
+	Sev  uint8  `json:"Severity"`
+}
+
+// ServicelogParser implements the DBReader interface.
+type ServicelogParser struct {
+	dbPath                     string
+	severity                   int
+	currLogCount, prevLogCount int
+}
+
+func NewReader(dbFilePath string, eventSeverity int) *ServicelogParser {
+	return &ServicelogParser{
+		dbPath:   dbFilePath,
+		severity: eventSeverity,
+	}
+}
+
+// ReadServicelogDB accesses the servicelog.db and retrieves the rows, based on
+// the severity threshold configured. The retrieved entries are logged and published as
+// an event to the kubernetes API server.
+func (reader *ServicelogParser) ReadServicelogDB(notifier *knode.K8sNotifier) error {
```

Comment:
this function name sounded like only reads the event and also the package name is dbreader! so I see a scope for making the code better here.